### PR TITLE
No encodes

### PR DESCRIPTION
**Pull request by RainerBitfly**

### CODE REVIEW

File: db/db.go [hunk 1]
```diff
@@ -98,26 +98,41 @@ func GetEth1DepositsJoinEth2Deposits(query string, length, start uint64, orderBy
...
 			hasColumn = true
+			break
 		}
 	}
 	if !hasColumn {
 		orderBy = "block_ts"
 	}
 
-	var err error
+	var searchLikeHash = regexp.MustCompile(`^0?x?[0-9a-fA-F]{2,96}`) // only search for pubkeys if string consists of 96 hex-chars
```
Comment (qu0b) [inline db/db.go:108]: Please compile regex outside of the  handler/function

File: db/db.go [hunk 2]
```diff
@@ -156,14 +171,14 @@ func GetEth1DepositsJoinEth2Deposits(query string, length, start uint64, orderBy
 		ON
 			v.pubkey = eth1.publickey
 		WHERE
-			ENCODE(eth1.publickey::bytea, 'hex') LIKE LOWER($5)
-			OR ENCODE(eth1.withdrawal_credentials::bytea, 'hex') LIKE LOWER($5)
-			OR ENCODE(eth1.from_address::bytea, 'hex') LIKE LOWER($5)
-			OR ENCODE(tx_hash::bytea, 'hex') LIKE LOWER($5)
-			OR CAST(eth1.block_number AS text) LIKE LOWER($5)
+			eth1.publickey LIKE $5 || '%%'::bytea
```
Comment (qu0b) [inline db/db.go:174]: Why do you sometimes use one '%' and sometimes two '%%'?

File: handlers/user.go [hunk 3]
```diff
@@ -328,10 +328,10 @@ func getValidatorTableData(userId uint64) (interface{}, error) {
 	}{}
 
 	err := db.FrontendDB.Select(&validatordb, `
-SELECT ENCODE(uvt.validator_publickey::bytea, 'hex') AS pubkey, us.event_name, extract( epoch from last_sent_ts)::Int as last_sent_ts, us.event_threshold
-FROM users_validators_tags uvt
-LEFT JOIN users_subscriptions us ON us.event_filter = ENCODE(uvt.validator_publickey::bytea, 'hex') AND us.user_id = uvt.user_id
-WHERE uvt.user_id = $1;`, userId)
+	SELECT ENCODE(uvt.validator_publickey::bytea, 'hex') AS pubkey, us.event_name, extract( epoch from last_sent_ts)::Int as last_sent_ts, us.event_threshold
+		FROM users_validators_tags uvt
+		LEFT JOIN users_subscriptions us ON us.event_filter = uvt.validator_publickey::bytea AND us.user_id = uvt.user_id
```
Comment (qu0b) [inline handlers/user.go:333]: event_filter is a string and you are comparing it to a bytea `us.event_filter = uvt.validator_publickey::bytea`. I don't think this change is necessary.

**Review by qu0b — CHANGES_REQUESTED** [review]

File: db/db.go (hunk 1 above)
Comment (RainerBitfly) [reply db/db.go]: Did it in the last commit.

File: handlers/user.go (hunk 3 above)
Comment (RainerBitfly) [reply handlers/user.go:333]: Did no changes in this row

**Review by qu0b — APPROVED** [review]

**Pull request merged**